File: javascript/resistor-color-duo/resistor-color-duo.js
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
];

export const value = ([color1, color2]) =>
  parseInt(`${COLORS.indexOf(color1)}${COLORS.indexOf(color2)}`);

File: ruby/resistor-color-duo/resistor_color_duo.rb
RESISTOR_COLOR = {
  'Black'=> 0,
  'Brown'=> 1,
  'Red'=> 2,
  'Orange'=> 3,
  'Yellow'=> 4,
  'Green'=> 5,
  'Blue'=> 6,
  'Violet'=>7,
  'Grey'=> 8,
  'White'=> 9
}

module ResistorColorDuo  
  def self.value(array)
    array.map do |x|
      x = RESISTOR_COLOR[x.capitalize].to_s
    end.join.to_i
  end 
end
